Gombang/[LeetCode75] - [Stack]/735_Asteroid_Collision.cpp

```diff
@@ -0,0 +1,56 @@
+class Solution {
+public:
+    vector<int> asteroidCollision(vector<int>& asteroids) {
+        stack<int> asteroidStack;
+        for (int newAsteroid : asteroids)
+        {
+            while (true)
+            {
+                if (asteroidStack.empty())
+                {
+                    asteroidStack.push(newAsteroid);
+                    break;
+                }
+
+                int topAsteroid = asteroidStack.top();
+
+                // 서로 같은 방향 또는 충돌이 일어나지 않는 방향이면 충돌하지 않는다.
+                if ((topAsteroid > 0 && newAsteroid > 0) ||
+                    (topAsteroid < 0 && newAsteroid < 0) ||
+                    (topAsteroid < 0 && newAsteroid > 0))
```

Comment:
서로 같은 방향일 때, 조건을 곱하기 연산을 사용해서 하나로 합쳐보는 것도 조건을 줄이는 방법일 것 같아서 코멘트 달아봅니다~
```suggestion
                    if(topAsteroid * newAsteroid > 0 || (topAsteroid < 0 && newAsteroid  > 0))
```